docs/docs/developers/development/dialogs/index.md

```diff
@@ -64,4 +70,105 @@ It will result with following in UI dialog:
 
 ![Dialog example tab2](dialog-example-tab2.png)
 
-Explore available fields and options.
\ No newline at end of file
+## Data structure
+All fields within the dialog have the `name` property, which defines the property’s name, under which values are saved in JCR. What is significant, the data is flattened, which means it does not matter if a particular field is in a nested structure (like tabs) or not. Properties are always saved under the same node.
+
+## Validation
+Validation in dialog fields is used to verify if the value meets the criteria of the particular field. There are two levels:
+
+### FrontEnd validation
+Front End validation is done by Atlaskit by default, if the submitted value is obviously incorrect, like a non-number value in NumberField. You can find more details [here](https://atlassian.design/components).
+![](dialog-frontend-validation.png)
+
+### BackEnd validation
+WebSight supports the validation of dialog values on BackEnd side. If the value is incorrect, then it won’t be saved and the dialog can’t be submitted. The author will see a proper error message.
+![](dialog-backend-validation.png)
+
+## Show/hide dialog fields
+By default, all dialog components are visible, but there is a possibility to hide them.
+
+### Context
+To show or hide a particular field depending on dialog context you can use a `ws:disallowedContext` parameter.
+
+```json
+"ws:disallowedContext": ["edit"]
+```
+
+To hide an element in dialog, the request from Front-End which fetches it has to contain the additional parameter `context`. If the context value matches one of `ws:dissallowedContext` values, then the field won’t be rendered.
```

Comment:
add link to swagger browser tool with rest action api <instace>/apps/apidocs#/apps/websight-dialogs-service/docs/api.html